data-prepper-script/src/main/java/org/opensearch/dataprepper/script/parser/LogListener.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.script.parser;
+
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.tree.ErrorNode;
+import org.antlr.v4.runtime.tree.TerminalNode;
+import org.opensearch.dataprepper.script.antlr.DataPrepperScriptListener;
+import org.opensearch.dataprepper.script.antlr.DataPrepperScriptParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * @since 1.3
+ * LogListener is a utility listener that logs every event enter and exit. Useful for debugging and developing new
+ * listeners.
```

Comment:
This seems beyond the scope of the PR. What is the benefit of this logger over using a debugger? Might I propose, we separate the LogListeners out of this PR?